src/Compilers/Core/Portable/Text/StringBuilderText.cs

```diff
@@ -89,5 +91,45 @@ public override void CopyTo(int sourceIndex, char[] destination, int destination
         {
             _builder.CopyTo(sourceIndex, destination, destinationIndex, count);
         }
+
+        public override void Write(TextWriter textWriter, TextSpan span, CancellationToken cancellationToken = default)
+        {
+#if NETCOREAPP
+            if (span.Start == 0 && span.Length == this.Length)
+            {
+                textWriter.Write(_builder);
+            }
```

Comment:
nit, consider doing an early return after the trivial case, that way it's clear nothing follows it.  it also enables the complex case to not need indentation.

Reply:
Done.